### PR TITLE
Clockwork scripture now uses one progress bar instead of multiple

### PR DESCRIPTION
**Pull request by ghost**
:cl: Xhuis
refactor: Clockwork scripture now has one progress bar for the entire recital instead of multiple ones for each sentence in the invocation.
/:cl:

This is a small but significant polish thing that's been in clockcult since its conception. Since seeing a progress bar complete and then seeing another progress bar is very strange, I moved invocation to a new `chant()` proc that says them at the same times, but now the scripture has one progress bar.

### CODE REVIEW

**Comment by Supermichael777** [comment]
Thank fuck this was irritating

File: code/game/gamemodes/clock_cult/clock_scripture.dm [hunk 1]
```diff
@@ -12,6 +12,7 @@ Applications: 8 servants, 3 caches, and 100 CV
 	var/name = "scripture"
 	var/desc = "Ancient Ratvarian lore. This piece seems particularly mundane."
 	var/list/invocations = list() //Spoken over time in the ancient language of Ratvar. See clock_unsorted.dm for more details on the language and how to make it.
+	var/chanting //If the invocation's words are being spoken
```
Comment (ChangelingRain) [inline code/game/gamemodes/clock_cult/clock_scripture.dm]: do not initialize numerical vars as null

Comment (Xhuis) [reply code/game/gamemodes/clock_cult/clock_scripture.dm]: It's a boolean; it only needs to check if it exists or not.

Comment (ChangelingRain) [reply code/game/gamemodes/clock_cult/clock_scripture.dm]: `var/chanting = FALSE`

Comment (Xhuis) [reply code/game/gamemodes/clock_cult/clock_scripture.dm]: That doesn't change my point, and two other variables in this exact datum (`invoker` and `slab`, which also check for existence) function in the exact same way.

Comment (ChangelingRain) [reply code/game/gamemodes/clock_cult/clock_scripture.dm]: those aren't booleans.
`var/chanting = FALSE`

**Comment by AnonymousNow** [comment]
Great for readability. Chanting makes you very vulnerable, and being able to tell how long it'd take is important - multiple bars is unusual, and makes it easy to accidentally cancel out during a later phase.

File: code/game/gamemodes/clock_cult/clock_scripture.dm [hunk 2]
```diff
@@ -145,18 +146,29 @@ Applications: 8 servants, 3 caches, and 100 CV
...
+/datum/clockwork_scripture/proc/chant()
+	set waitfor = FALSE
+	chanting = TRUE
 	for(var/invocation in invocations)
-		if(!do_after(invoker, channel_time / invocations.len, target = invoker, extra_checks = CALLBACK(src, .proc/check_special_requirements)))
-			slab.busy = null
-			scripture_fail()
-			return FALSE
+		sleep(channel_time / invocations.len)
+		if(!chanting)
```
Comment (Cyberboss) [inline code/game/gamemodes/clock_cult/clock_scripture.dm]: check QDELETED(src) first

**Review by Cyberboss — CHANGES_REQUESTED** [review]

**Pull request merged**